File: _5_replace_0_with_1.java


public class _5_replace_0_with_1{
    public static void main(String[] args){

        int N = 1301032;
        int ans = 0;
        int i = 0;

        while(N != 0){
            if(N%10 == 0){
                ans = (int) (ans + 1 * Math.pow(10, i));
            }
            else{
                ans =  (int) (ans + (N%10) * Math.pow(10,i));
            }

            N = N / 10;
            i++;
        }

        System.out.println(ans);

    }
}

File: cpp/5_replace_0_with_1.cpp
#include<iostream>
#include<math.h>
using namespace std;

int main(){

    int N = 10106570;
    int ans = 0;
    int i = 0;

    while(N != 0){
        if((N%10) == 0){
            ans = ans + 1 * pow(10,i);
        }
        else{
            ans = ans + (N%10) * pow(10,i);
        }
        N = N/10;
        i++;
    }

    cout<<ans;

}